src/main/java/seedu/address/logic/commands/TagCommand.java

```diff
@@ -23,59 +24,95 @@ public class TagCommand extends Command {
+            + ": Tags the guest identified by the index number used in the displayed guest list "
+                + "with predefined tag(s). \n"
             + "Parameters: INDEX (must be a positive integer)\n"
-            + "[" + PREFIX_TAG + "TAG] (must be defined using 'newtag' command first\n"
+            + "[" + PREFIX_TAG + "TAG]... (must be created using 'newtag' command first)\n"
             + "Example: " + COMMAND_WORD + " 1 "
             + PREFIX_TAG + "bride's side";
 
-    public static final String MESSAGE_TAG_PERSON_SUCCESS = "Tagged Person: %1$s";
-    public static final String MESSAGE_TAG_NOT_CREATED = "Tag must be created first using (newtag) command. \n";
-    public static final String MESSAGE_DUPLICATE_TAG = "This person already has this tag.";
+    public static final String MESSAGE_TAG_PERSON_SUCCESS = "Tagged guest(s):\n";
+    public static final String MESSAGE_TAG_NOT_CREATED = "Tag(s) must be created first using 'newtag' command: ";
+    public static final String MESSAGE_DUPLICATE_TAG = "Some guest(s) already have tag(s): ";
 
-    private final Index targetIndex;
-    private final Tag tag;
+    private final List<Index> targetIndexes;
+    private final Set<Tag> tags;
 
     /**
-     * @param targetIndex of the person in the filtered person list to tag
-     * @param tag to tag the person with
+     * @param targetIndexes of the person in the filtered person list to tag
+     * @param tags set of tags to tag the person with
      */
-    public TagCommand(Index targetIndex, Tag tag) {
-        requireNonNull(targetIndex);
-        requireNonNull(tag);
+    public TagCommand(List<Index> targetIndexes, Set<Tag> tags) {
+        requireNonNull(targetIndexes);
+        requireNonNull(tags);
 
-        this.targetIndex = targetIndex;
-        this.tag = tag;
+        this.targetIndexes = targetIndexes;
+        this.tags = tags;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        Set<Tag> missingTags = new HashSet<>();
+        Set<Tag> duplicateTags = new HashSet<>();
+        StringBuilder successMessage = new StringBuilder();
+        StringBuilder finalMessage = new StringBuilder();
+
+        for (Index targetIndex : targetIndexes) {
+            if (targetIndex.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+            Person personToTag = lastShownList.get(targetIndex.getZeroBased());
+            Set<Tag> newTags = new HashSet<>(personToTag.getTags());
+            boolean updated = false;
+
+            for (Tag tag : tags) {
+                if (!model.hasTag(tag)) {
```

Comment:
Can the checking be extracted out into a method to improve SLAP?

Reply:
I feel like it's abit unnecessary to abstract out into a method